### DISCUSSION TITLE
[v1.0.0-rc3] Splash screen jumps on initial load (android prebuild)

### DISCUSSION THREAD

**Discussion by jjenzz** [expo/router#595] [Q&A]
### Summary

when moving to prebuild (`npx expo prebuild`) from a fresh install (`npx create-react-native-app -t with-router`) the splash screen will jump up for one frame on android during app load. it also still occurs when upgrading to `1.0.0-rc3`.

you can slow the following video frame-by-frame to see what i mean:

https://user-images.githubusercontent.com/175330/215787694-9762836b-1b74-4d06-bd3b-ad1de3b76b05.mp4

the two frames where the jump occurs (i've drawn a redline to highlight the difference in position):

![image](https://user-images.githubusercontent.com/175330/215788472-6b6b4126-90c4-4e97-8330-f15c754fb403.png)

while it is hard to catch this with the naked eye in this demo, the jump is much more prominent in our real-world app that takes a little longer to load.

i'm not entirely sure if this is a router issue or more generic expo issue so happy to move it to the correct place if there is somewhere more suitable.

### Minimal reproducible example

https://github.com/jjenzz/expo-splash-jump

**Comment by EvanBacon** [comment]
It's most likely not an Expo Router issue. I'd imagine the problem is related to the status bar being "translucent" (ie. position: absolute), this causes the quick shift upward. https://docs.expo.dev/versions/latest/config/app/#translucent

**Comment by jjenzz** [comment]
to clarify, this issue doesn't occur in a managed workflow, simply moving from managed to bare introduces it afaict. 

i have not rendered a `StatusBar` component anywhere in linked source and haven't configured one in `app.json` so i am curious what difference moving to prebuild could have here? does it make the status bar translucent? 

EDIT: i've recreated this issue in the expo repo as it sounds like that is a more suitable place for it. lmk if not :)

**Comment by EvanBacon** [comment]
@jjenzz you can try removing the [StatusBar component that is rendered in Expo Router](https://github.com/expo/router/blob/11c8b94718ef7c37accc5099654b501912bfac4f/packages/expo-router/src/views/Root.tsx#L34) but it should be benign.

There is no bare/managed anymore (we need to improve our docs on this). "Managed" is now bare + prebuild. You can either manually run prebuild, or use EAS Build which runs prebuild to generate the native code required for building meaning there should be virtually no difference (outside of minor build-time changes in Expo Go).

The previous "managed workflow" would embed the JS in a shell application which would never match the "bare workflow" variation, we fully removed this by migrating to EAS Build.

**Comment by jjenzz** [comment]
i am perhaps confusing things by using the wrong terminology in that case, apologies. but what i mean is that `npx create-react-native-app -t with-router` and then `npm run android` does not have the issue, but as soon as i run `npx expo prebuild` to create the `ios` and `android` folders, an `npm run android` **will** have this issue.

thanks for the status bar suggestion, i will give that a try.

**Comment by matuskropuch** [comment]
For me, removing the StatusBar as you have suggested fixes the jump.

**Comment by andreavrr** [comment]
Other solutions without hiding the status bar?
